src/cider/exec/nextgen/context/CodegenContext.h

```diff
@@ -92,16 +114,41 @@ class CodegenContext {
         : ctx_id(id), name(n), capacity(c), initializer_(initializer) {}
   };
 
+  struct HashTableDescriptor {
+    int64_t ctx_id;
+    std::string name;
+    // TODO(qiuyang) : LinearProbeHashTable will provide the default template
+    LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>* hash_table;
```

Comment:
I suggest to define and use base class pointer here. Otherwise we may have huge change(or redesign) in the future. cc @spevenhe